packages/emitter/src/index.ts

```diff
@@ -25,12 +45,33 @@ function escapeQuotes(value: string, quote: Quote) {
   return value.replace(NEEDS_ESCAPING[quote], "\\$&");
 }
 
-function escapeValue(value: string, quote: Quote) {
-  if (value === "") {
-    return quote + quote;
+const QUOTE_COUNTER: { [Q in Quote]: RegExp } = {
+  '"': /"/g,
+  "'": /'/g,
+};
+
+function countQuote(value: string, quote: Quote) {
+  return value.match(QUOTE_COUNTER[quote])?.length ?? 0;
+}
+
+function selectQuote(
+  value: string,
+  {
+    preferredQuote = DEFAULT_EMIT_OPTIONS.preferredQuote,
+    optimizeQuotes = DEFAULT_EMIT_OPTIONS.optimizeQuotes,
+  }: EmitOptions
+) {
+  if (optimizeQuotes) {
+    const otherQuote: Quote = preferredQuote === '"' ? "'" : '"';
+    return countQuote(value, otherQuote) < countQuote(value, preferredQuote) ? otherQuote : preferredQuote;
+  } else {
+    return preferredQuote;
   }
+}
 
-  if (ReservedChars.some((reservedChar) => value.includes(reservedChar))) {
+function escapeValue(value: string, opts: EmitOptions) {
```

Comment:
nit: could we use `options` instead of `opts` name? I just prefer to use full names and I want to be consistent :)